### PR TITLE
Added typing for string-placeholder

### PR DESCRIPTION
**Pull request by fbritoferreira**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
-  [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/string-placeholder/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for string-placeholder 1.0
+// Project: https://github.com/crysalead-js/string-placeholder#readme
+// Definitions by: Filipe Ferreira <https://github.com/iTsFILIPOficial>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.1
+
+export = template;
+export as namespace template;
```
Comment (PranavSenthilnathan) [inline types/string-placeholder/index.d.ts]: I don't this module exports a global so this line isn't needed.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/string-placeholder/string-placeholder-tests.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import template from 'string-placeholder';
+
+// $ExpectType string
+template('{0} test', ['here']);
+
```
Comment (demensky) [inline types/string-placeholder/string-placeholder-tests.ts]: Please add a case with an object.

```suggestion

template('${foo} test ${bar}', {foo: 'Bob', bar: '65'});

```

**Review by demensky — CHANGES_REQUESTED** [review]

File: types/string-placeholder/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.1
+
+export function template(str: string, data: Readonly<[]> | Readonly<{}>, options?: Options): string;
+
+interface Options {
+    before?: string;
+    after?: string;
+    escape?: string;
+    clean?: boolean | Readonly<[]>;
```
[the remark this answers is not in the document]
Comment (iTsFILIPOficial) [reply types/string-placeholder/index.d.ts]: @demesky https://github.com/crysalead-js/string-placeholder/blob/master/index.js#L25

[the remark this answers is not in the document]
Comment (demensky) [reply types/string-placeholder/index.d.ts]: @iTsFILIPOficial. I agree with the comments mentioned array. But in the source code, it is used only as a boolean value. I can’t even find a spec where an array is passed to `clean`.

**Review by demensky — APPROVED** [review]

File: types/string-placeholder/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export function template(str: string, data: Readonly<unknown>, options?: Readonly<Options>): string;
+
+interface Options {
+    before?: string;
+    after?: string;
+    escape?: string;
+    clean?: boolean;
+}
+
+export default template;
```
Comment (PranavSenthilnathan) [inline types/string-placeholder/index.d.ts]:
```suggestion
export = template;
```

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

**Review by PranavSenthilnathan — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**